dashboard/lib/services/user/password_resetter_by_username.rb

```diff
@@ -0,0 +1,34 @@
+module Services
+  module User
+    class PasswordResetterByUsername < Services::Base
+      attr_reader :username
+
+      def initialize(username: nil)
+        @username = username
+      end
+
+      def call
+        user = ::User.find_by(username: username) || ::User.new(username: username)
+        return user if user.errors.present?
+
+        if user.new_record?
+          Cdo::Metrics.put(
+            'User', 'PasswordResetUserNotFound', 1, {
+              Environment: CDO.rack_env
+            }
+          )
+          return user
+        end
+
+        # Just print out password reset instructions for Admin user
+        user.raw_token = user.send(:set_reset_password_token)
```

Comment:
Why don't we save the user record after assigning the `raw_token`?

Reply:
If we trace it back, this is called by [this method](https://github.com/heartcombo/devise/blob/main/app/controllers/devise/passwords_controller.rb#L14-L23) in Devise, so we set the user as the user returned by this call method.